When Tests are Run in `tox` on Windows, the `Printer` Tests Fail

### Describe the bug

Running `tox -e py -- tests\console\test_Printer.py` on Windows in PowerShell or CMD fails. (This failure doesn't happen in git bash.)

```
py: commands[0]> python -m coverage run -m pytest tests\console\test_Printer.py
================================================= test session starts =================================================
platform win32 -- Python 3.11.3, pytest-7.3.1, pluggy-1.0.0
cachedir: .tox\py\.pytest_cache
rootdir: C:\Users\user\github\beeware\briefcase
configfile: pyproject.toml
plugins: xdist-3.2.1
collected 5 items

tests\console\test_Printer.py FF.FF                                                                              [100%]

====================================================== FAILURES =======================================================
_________________________________ test_call[True-a line of output-a line of output\n] _________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x0000024FE79D6D50>
printer = <briefcase.console.Printer object at 0x0000024FE79E74D0>, message = 'a line of output', show = True
expected_console_output = 'a line of output\n'

    @pytest.mark.parametrize(
        "show, message, expected_console_output",
        [
            (True, "a line of output", "a line of output\n"),
            (False, "a line of output", ""),
        ],
    )
    def test_call(capsys, printer, message, show, expected_console_output):
        """Printer prints to console and log appropriately."""
        printer(message, show=show, stack_offset=1)
        assert capsys.readouterr().out == expected_console_output
        log = printer.export_log()
        assert len(log.splitlines()) == 1
>       assert " " + message + " " * 139 + "console.py" in log
E       AssertionError: assert (((' ' + 'a line of output') + (' ' * 139)) + 'console.py') in '[20:49:40] a line of output
                    console.py:108\n'

tests\console\test_Printer.py:28: AssertionError
_________________________________________ test_call[False-a line of output-] __________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x0000024FE79FAA50>
printer = <briefcase.console.Printer object at 0x0000024FE79FBE90>, message = 'a line of output', show = False
expected_console_output = ''

    @pytest.mark.parametrize(
        "show, message, expected_console_output",
        [
            (True, "a line of output", "a line of output\n"),
            (False, "a line of output", ""),
        ],
    )
    def test_call(capsys, printer, message, show, expected_console_output):
        """Printer prints to console and log appropriately."""
        printer(message, show=show, stack_offset=1)
        assert capsys.readouterr().out == expected_console_output
        log = printer.export_log()
        assert len(log.splitlines()) == 1
>       assert " " + message + " " * 139 + "console.py" in log
E       AssertionError: assert (((' ' + 'a line of output') + (' ' * 139)) + 'console.py') in '           a line of output
                    console.py:108\n'

tests\console\test_Printer.py:28: AssertionError
_____________________________________________________ test_to_log _____________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x0000024FE7A52790>
printer = <briefcase.console.Printer object at 0x0000024FE7A50490>

    def test_to_log(capsys, printer):
        """Printer prints only to log."""
        printer.to_log("a line of output", stack_offset=1)
        assert capsys.readouterr().out == ""
        log = printer.export_log()
        assert len(log.splitlines()) == 1
>       assert " a line of output" + " " * 139 + "console.py" in log
E       AssertionError: assert ((' a line of output' + (' ' * 139)) + 'console.py') in '           a line of output     
             console.py:108\n'

tests\console\test_Printer.py:44: AssertionError
_________________________________________________ test_very_long_line _________________________________________________

capsys = <_pytest.capture.CaptureFixture object at 0x0000024FE7A56C50>
printer = <briefcase.console.Printer object at 0x0000024FE7A57250>

    def test_very_long_line(capsys, printer):
        """Very long lines are split."""
        printer.to_log("A very long line of output!! " * 6, stack_offset=1)
        assert capsys.readouterr().out == ""
        log = printer.export_log()
        assert len(log.splitlines()) == 2
>       assert (" " + "A very long line of output!! " * 5 + "A very    console.py:") in log
E       AssertionError: assert ((' ' + ('A very long line of output!! ' * 5)) + 'A very    console.py:') in '
A very long line of output!! A very long line of output!! A very long line of output!! A very long line of...
                                                                                                         \n'

tests\console\test_Printer.py:53: AssertionError
=============================================== short test summary info ===============================================
FAILED tests/console/test_Printer.py::test_call[True-a line of output-a line of output\n] - AssertionError: assert ((...
FAILED tests/console/test_Printer.py::test_call[False-a line of output-] - AssertionError: assert (((' ' + 'a line of...
FAILED tests/console/test_Printer.py::test_to_log - AssertionError: assert ((' a line of output' + (' ' * 139)) + 'co...
FAILED tests/console/test_Printer.py::test_very_long_line - AssertionError: assert ((' ' + ('A very long line of outp...
============================================= 4 failed, 1 passed in 0.12s =============================================
```

However, `pytest -vv tests\console\test_Printer.py` works fine...

```
❯ pytest -vv tests\console\test_Printer.py
platform win32 -- Python 3.11.3
pytest==7.3.1
pluggy==1.0.0
rootdir: C:\Users\user\github\beeware\briefcase
inifile: pyproject.toml
plugins: tldr-0.2.5, xdist-3.2.1
cachedir: .pytest_cache

------------------------------------------------------------------------------
tests/console/test_Printer.py::test_call[True-a line of output-a line of output\n] ... ok
tests/console/test_Printer.py::test_call[False-a line of output-] ... ok
tests/console/test_Printer.py::test_to_console ... ok
tests/console/test_Printer.py::test_to_log ... ok
tests/console/test_Printer.py::test_very_long_line ... ok

------------------------------------------------------------------------------
Ran 5 tests in 0.03s

OK
```

### Steps to reproduce

Run `tox -e py` on Windows in PowerShell or CMD.

### Expected behavior

Tests pass.

### Screenshots

_No response_

### Environment

- Operating System: Windows 11 22H2
- Python version: Python 3.11/10 (winstore or otherwise)
- Software versions:
  - Briefcase: `0.3.14`, `0.3.15.dev148+gda62b7c5`

### Logs

_No response_

### Additional context

_No response_

--- a/changes/1249.misc.rst
+++ b/changes/1249.misc.rst
@@ -1,0 +1,1 @@
+The ``Printer`` tests no longer fail on Windows when run in ``tox``.

--- a/docs/spelling_wordlist
+++ b/docs/spelling_wordlist
@@ -31,6 +31,7 @@ gtk
 hoc
 hostname
 installable
+iOS
 jarsigner
 keychain
 Keychain

--- a/docs/spelling_wordlist
+++ b/docs/spelling_wordlist
@@ -16,6 +16,7 @@ cryptographic
 dev
 dialogs
 distributable
+Django
 dmg
 Dockerfile
 embeddable

--- a/tests/console/test_Printer.py
+++ b/tests/console/test_Printer.py
@@ -1,3 +1,5 @@
+import re
+
 import pytest
 
 from briefcase.console import Printer
@@ -12,9 +14,13 @@ def printer():
     return printer
 
 
-def no_sp(text: str) -> str:
-    """Remove spaces from text."""
-    return text.replace(" ", "")
+def norm_sp(text: str) -> str:
+    """Normalize >100 spaces in text to 100 spaces.
+
+    This is necessary until https://github.com/Textualize/rich/issues/2944 is resolved
+    to ensure that tests succeed while running in `tox` on Windows.
+    """
+    return re.sub(r"\s{100,}", " " * 100, text)
 
 
 @pytest.mark.parametrize(
@@ -31,8 +37,7 @@ def test_call(capsys, printer, message, show, expected_console_output):
     log = printer.export_log()
     assert len(log.splitlines()) == 1
     # The number of spaces in not consistent on Windows.
-    # Remove `no_sp()` when https://github.com/Textualize/rich/issues/2944 is resolved
-    assert no_sp(" " + message + " " * 139 + "console.py") in no_sp(log)
+    assert norm_sp(" " + message + " " * 139 + "console.py") in norm_sp(log)
 
 
 def test_to_console(capsys, printer):
@@ -49,8 +54,7 @@ def test_to_log(capsys, printer):
     log = printer.export_log()
     assert len(log.splitlines()) == 1
     # The number of spaces in not consistent on Windows.
-    # Remove `no_sp()` when https://github.com/Textualize/rich/issues/2944 is resolved
-    assert no_sp(" a line of output" + " " * 139 + "console.py") in no_sp(log)
+    assert norm_sp(" a line of output" + " " * 139 + "console.py") in norm_sp(log)
 
 
 def test_very_long_line(capsys, printer):
@@ -60,8 +64,7 @@ def test_very_long_line(capsys, printer):
     log = printer.export_log()
     assert len(log.splitlines()) == 2
     # The number of spaces in not consistent on Windows.
-    # Remove `no_sp()` when https://github.com/Textualize/rich/issues/2944 is resolved
-    assert no_sp(
+    assert norm_sp(
         " " + "A very long line of output!! " * 5 + "A very    console.py:"
-    ) in no_sp(log)
-    assert no_sp(" long line of output!!" + " " * 148) in no_sp(log)
+    ) in norm_sp(log)
+    assert norm_sp(" long line of output!!" + " " * 148) in norm_sp(log)

--- a/tests/console/test_Printer.py
+++ b/tests/console/test_Printer.py
@@ -12,6 +12,11 @@ def printer():
     return printer
 
 
+def no_sp(text: str) -> str:
+    """Remove spaces from text."""
+    return text.replace(" ", "")
+
+
 @pytest.mark.parametrize(
     "show, message, expected_console_output",
     [
@@ -25,7 +30,9 @@ def test_call(capsys, printer, message, show, expected_console_output):
     assert capsys.readouterr().out == expected_console_output
     log = printer.export_log()
     assert len(log.splitlines()) == 1
-    assert " " + message + " " * 139 + "console.py" in log
+    # The number of spaces in not consistent on Windows.
+    # Remove `no_sp()` when https://github.com/Textualize/rich/issues/2944 is resolved
+    assert no_sp(" " + message + " " * 139 + "console.py") in no_sp(log)
 
 
 def test_to_console(capsys, printer):
@@ -41,7 +48,9 @@ def test_to_log(capsys, printer):
     assert capsys.readouterr().out == ""
     log = printer.export_log()
     assert len(log.splitlines()) == 1
-    assert " a line of output" + " " * 139 + "console.py" in log
+    # The number of spaces in not consistent on Windows.
+    # Remove `no_sp()` when https://github.com/Textualize/rich/issues/2944 is resolved
+    assert no_sp(" a line of output" + " " * 139 + "console.py") in no_sp(log)
 
 
 def test_very_long_line(capsys, printer):
@@ -50,5 +59,9 @@ def test_very_long_line(capsys, printer):
     assert capsys.readouterr().out == ""
     log = printer.export_log()
     assert len(log.splitlines()) == 2
-    assert (" " + "A very long line of output!! " * 5 + "A very    console.py:") in log
-    assert (" long line of output!!" + " " * 148) in log
+    # The number of spaces in not consistent on Windows.
+    # Remove `no_sp()` when https://github.com/Textualize/rich/issues/2944 is resolved
+    assert no_sp(
+        " " + "A very long line of output!! " * 5 + "A very    console.py:"
+    ) in no_sp(log)
+    assert no_sp(" long line of output!!" + " " * 148) in no_sp(log)

--- a/tests/console/test_Printer.py
+++ b/tests/console/test_Printer.py
@@ -14,13 +14,13 @@ def printer():
     return printer
 
 
-def norm_sp(text: str) -> str:
-    """Normalize >100 spaces in text to 100 spaces.
+def norm_sp(text: str, max_spaces: int = 100) -> str:
+    """Normalize more than `max_spaces` spaces in text to `max_spaces` spaces.
 
     This is necessary until https://github.com/Textualize/rich/issues/2944 is resolved
     to ensure that tests succeed while running in `tox` on Windows.
     """
-    return re.sub(r"\s{100,}", " " * 100, text)
+    return re.sub(rf"\s{{{max_spaces},}}", " " * max_spaces, text)
 
 
 @pytest.mark.parametrize(
@@ -36,7 +36,7 @@ def test_call(capsys, printer, message, show, expected_console_output):
     assert capsys.readouterr().out == expected_console_output
     log = printer.export_log()
     assert len(log.splitlines()) == 1
-    # The number of spaces in not consistent on Windows.
+    # The number of spaces in not consistent on Windows
     assert norm_sp(" " + message + " " * 139 + "console.py") in norm_sp(log)
 
 
@@ -53,7 +53,7 @@ def test_to_log(capsys, printer):
     assert capsys.readouterr().out == ""
     log = printer.export_log()
     assert len(log.splitlines()) == 1
-    # The number of spaces in not consistent on Windows.
+    # The number of spaces in not consistent on Windows
     assert norm_sp(" a line of output" + " " * 139 + "console.py") in norm_sp(log)
 
 
@@ -63,8 +63,9 @@ def test_very_long_line(capsys, printer):
     assert capsys.readouterr().out == ""
     log = printer.export_log()
     assert len(log.splitlines()) == 2
-    # The number of spaces in not consistent on Windows.
+    # The number of spaces in not consistent on Windows
     assert norm_sp(
-        " " + "A very long line of output!! " * 5 + "A very    console.py:"
-    ) in norm_sp(log)
+        " " + "A very long line of output!! " * 5 + "A very    console.py:",
+        max_spaces=3,
+    ) in norm_sp(log, max_spaces=3)
     assert norm_sp(" long line of output!!" + " " * 148) in norm_sp(log)

